onsstove/onsstove.py

```diff
@@ -567,10 +567,22 @@ def save_datasets(self, datasets='all'):
                                            category, name)
                 layer.save(output_path)
 
-    def maximum_net_benefit(self, df):
+    def maximum_net_benefit(self):
+
+        net_benefit_cols = [col for col in self.gdf if 'final_tech' in col]
+        self.gdf["final_tech"] = self.gdf[net_benefit_cols].idxmin(axis=1)
+        self.gdf["maximum_net_benefit"] = self.gdf[net_benefit_cols].min(axis=1)
+
+        self.gdf['final_tech'] = self.gdf['final_tech'].str.replace("net_benefit_","")
+
+    def lives_saved(self):
+
+        total_pop = self.gdf['Calibrated_pop'].sum()
+
+        deaths_avoided = self.gdf.apply(lambda row: self.techs[row['final_tech']].deaths_avoided * row['Calibrated_pop'] / total_pop)
+
+        self.gdf["deaths_avoided"] = deaths_avoided.values
```

Comment:
I think in here we don't need the `.values` as this series should have the same indices as the main gdf. This is because the series was created using information from the gdf (`'Calibrated_pop'`)